common/note/note.go

```diff
@@ -1,43 +1,44 @@
 package note
 
 import (
+	"bytes"
 	"crypto/rand"
 	"crypto/sha256"
 	"encoding/binary"
 	"fmt"
 	"strings"
 
+	"github.com/ark-network/ark/common"
+	"github.com/ark-network/ark/common/bip322"
+	"github.com/ark-network/ark/common/tree"
 	"github.com/btcsuite/btcd/btcutil/base58"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
 )
 
-const noteHRP = "arknote"
+const (
+	preimageSize            = 32
+	noteHRP                 = "arknote"
+	fakeOutpointOutputIndex = uint32(0)
+)
 
-// Note represents a note signed by the issuer
+// Note contains the data of a note
 type Note struct {
-	Data
-	Signature []byte
-}
-
-// Data contains the data of a note
-type Data struct {
-	ID    uint64
-	Value uint32
+	Preimage []byte
```

Comment:
can be fixed length slice of 32 bytes
```suggestion
	Preimage [32]byte
```